src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php

```diff
@@ -69,14 +69,23 @@ public function registerContainerConfiguration(LoaderInterface $loader)
                 ],
             ]);
 
-            if ($this instanceof EventSubscriberInterface) {
+            if (!$container->hasDefinition('kernel')) {
                 $container->register('kernel', static::class)
                     ->setSynthetic(true)
                     ->setPublic(true)
-                    ->addTag('kernel.event_subscriber')
                 ;
             }
 
+            $kernelDefinition = $container->getDefinition('kernel');
+
+            $kernelDefinition->addTag('routing.router_loader', [
+                'service_id' => 'kernel',
```

Comment:
Do we have existing terminology for an attribute referencing a service?
What about using "id"?

Reply:
Tagged locators being new, we don't currently have usages in the core. We can use "id" here for sure, but the names mights vary depending on the tags and what they do I guess. Multiples services could be referenced for example.

Reply:
Tagged locators are a new way to reference tags, but they don't inherently allow building anything new.
Everything the provide is and was possible using compiler passes.
The practice of using attributes to reference ids is not new AFAIK - or is nothing surprising even with 2.0 :)
BUT this made me realize #31463 would be a better behavior that this PR could the leverage.

Reply:
I actually made the mistake to look only for tagged locators, not in compiler passes. Thank you for your explanations.  https://github.com/symfony/symfony/pull/31463 is a great idea! I will update the code here accordingly.